File: pophory-domain/src/main/java/com/pophory/pophorydomain/photo/infrastructure/PhotoJpaRepository.java
package com.pophory.pophorydomain.photo.infrastructure;

import com.pophory.pophorydomain.album.Album;
import com.pophory.pophorydomain.photo.Photo;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface PhotoJpaRepository extends JpaRepository<Photo, Long> {

    List<Photo> findAllByAlbum(Album album);

    Optional<Photo> findByShareId(String shareId);

}

File: pophory-domain/src/main/kotlin/com/pophory/pophorydomain/photo/infrastructure/PhotoJpaRepository.kt
package com.pophory.pophorydomain.photo.infrastructure

import com.pophory.pophorydomain.album.Album
import com.pophory.pophorydomain.photo.Photo
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface PhotoJpaRepository : JpaRepository<Photo, Long> {
    fun findAllByAlbum(album: Album?): List<Photo?>?

    fun findByShareId(shareId: String?): Optional<Photo?>?
}